package/samplers/nsgaii_with_initial_trials/nsgaii_with_initial_trials.py

```diff
@@ -0,0 +1,385 @@
+            )
+        )
+        self._child_generation_strategy = (
+            child_generation_strategy
+            or NSGAIIChildGenerationStrategy(
+                crossover_prob=crossover_prob,
+                mutation_prob=mutation_prob,
+                swapping_prob=swapping_prob,
+                crossover=crossover,
+                constraints_func=constraints_func,
+                rng=self._rng,
+            )
+        )
+        self._after_trial_strategy = after_trial_strategy or NSGAIIAfterTrialStrategy(
+            constraints_func=constraints_func
+        )
+
+    def reseed_rng(self) -> None:
+        self._random_sampler.reseed_rng()
+        self._rng.rng.seed()
+
+    def infer_relative_search_space(
+        self, study: Study, trial: FrozenTrial
+    ) -> dict[str, BaseDistribution]:
+        search_space: dict[str, BaseDistribution] = {}
+        for name, distribution in self._search_space.calculate(study).items():
+            if distribution.single():
+                # The `untransform` method of `optuna._transform._SearchSpaceTransform`
+                # does not assume a single value,
+                # so single value objects are not sampled with the `sample_relative` method,
+                # but with the `sample_independent` method.
+                continue
+            search_space[name] = distribution
+        return search_space
+
+    def sample_relative(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        search_space: dict[str, BaseDistribution],
+    ) -> dict[str, Any]:
+        parent_generation, parent_population = self._collect_parent_population(study)
+
+        generation = parent_generation + 1
+        study._storage.set_trial_system_attr(trial._trial_id, _GENERATION_KEY, generation)
+
+        if parent_generation < 0:
+            return {}
+
+        return self._child_generation_strategy(study, search_space, parent_population)
+
+    def sample_independent(
+        self,
+        study: Study,
+        trial: FrozenTrial,
+        param_name: str,
+        param_distribution: BaseDistribution,
+    ) -> Any:
+        # Following parameters are randomly sampled here.
+        # 1. A parameter in the initial population/first generation.
+        # 2. A parameter to mutate.
+        # 3. A parameter excluded from the intersection search space.
+
+        return self._random_sampler.sample_independent(
+            study, trial, param_name, param_distribution
+        )
+
+    def _collect_parent_population(self, study: Study) -> tuple[int, list[FrozenTrial]]:
+        trials = study._get_trials(deepcopy=False, use_cache=True)
+
+        generation_to_runnings = defaultdict(list)
+        generation_to_population = defaultdict(list)
+        for trial in trials:
+            if _GENERATION_KEY not in trial.system_attrs:
+                generation = 0
+            else:
+                generation = trial.system_attrs[_GENERATION_KEY]
```

Comment:
Only the treatment of the initial generation is changed.

However, I understand that this change may cause problems when the initial generation computing is parallelized.